tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs

```diff
@@ -15,41 +15,45 @@ namespace Datadog.Trace.Agent.DiscoveryService
 {
     internal class DiscoveryService : IDiscoveryService
     {
+        private const string InfoPath = "/info";
         private static readonly string[] SupportedDebuggerEndpoints = new[] { "debugger/v1/input" };
         private static readonly string[] SupportedProbeConfigurationEndpoints = new[] { "v0.7/config" };
+        private static readonly string[] SupportedStatsEndpoints = new[] { "v0.6/stats" };
 
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<DiscoveryService>();
 
         private readonly IApiRequestFactory _apiRequestFactory;
         private readonly Uri _agentUri;
 
         private DiscoveryService(
-            Uri agentUri,
             IApiRequestFactory apiRequestFactory)
         {
-            _agentUri = agentUri;
             _apiRequestFactory = apiRequestFactory;
+            _agentUri = apiRequestFactory.GetEndpoint(InfoPath);
         }
 
-        public static string[] AllSupportedEndpoints => SupportedDebuggerEndpoints.Concat(SupportedProbeConfigurationEndpoints).ToArray();
+        public static string[] AllSupportedEndpoints => SupportedDebuggerEndpoints.Concat(SupportedProbeConfigurationEndpoints).Concat(SupportedStatsEndpoints).ToArray();
 
         public string ProbeConfigurationEndpoint { get; private set; }
 
         public string DebuggerEndpoint { get; private set; }
 
+        public string StatsEndpoint { get; private set; }
+
         public string AgentVersion { get; private set; }
 
-        public static DiscoveryService Create(IConfigurationSource configurationSource, IApiRequestFactory apiRequestFactory)
+        public bool ClientDropP0s { get; private set; }
```

Comment:
If we don't have a retry mechanism, shall we make this a nullable to know we don't know?

Reply:
Yes good point. That's implemented in the new PR #3152